Dockerfile

```diff
@@ -44,21 +51,35 @@ RUN \
   dovecot-core dovecot-imapd dovecot-ldap dovecot-lmtpd \
   dovecot-managesieved dovecot-pop3d dovecot-sieve dovecot-solr \
   dumb-init \
-  # E - O
-  ed fetchmail file gamin gnupg gzip iproute2 iptables \
-  locales logwatch lhasa libdate-manip-perl liblz4-tool \
+  # # E - O
+  ed fetchmail file gamin gnupg gzip gpg=2.2.12-1+deb10u1 \
+  gpg-agent=2.2.12-1+deb10u1 inotify-tools \
+  iptables locales logwatch lhasa libdate-manip-perl liblz4-tool \
   libmail-spf-perl libnet-dns-perl libsasl2-modules lrzip lzop \
   netcat-openbsd nomarch opendkim opendkim-tools opendmarc \
   # P - Z
   pax pflogsumm postgrey p7zip-full postfix-ldap postfix-pcre \
   postfix-policyd-spf-python postsrsd pyzor \
   razor rpm2cpio rsyslog sasl2-bin spamassassin supervisor \
-  unrar-free unzip whois xz-utils >/dev/null && \
+  unrar-free unzip wget=1.20.1-1.1 whois xz-utils >/dev/null && \
   # Fail2Ban
-  curl -Lso fail2ban.deb https://github.com/fail2ban/fail2ban/releases/download/0.11.2/fail2ban_0.11.2-1.upstream1_all.deb && \
-  dpkg -i fail2ban.deb && \
-  rm fail2ban.deb && \
+  gpg --keyserver ${FAIL2BAN_PGP_PUBLIC_KEY_SERVER} \
+    --recv-keys ${FAIL2BAN_PGP_PUBLIC_KEY_ID} &>/dev/null && \
+  wget -q --no-check-certificate ${FAIL2BAN_URL} && \
+  wget -q --no-check-certificate ${FAIL2BAN_URL_ASC} && \
+  FINGERPRINT="$(LANG=C gpg --verify \
+    fail2ban_${FAIL2BAN_VERSION}-1.upstream1_all.deb.asc \
+    fail2ban_${FAIL2BAN_VERSION}-1.upstream1_all.deb 2>&1 \
+    | sed -n 's#Primary key fingerprint: \(.*\)#\1#p')" && \
+  if [[ -z ${FINGERPRINT} ]]; then \
+    echo "ERROR: Invalid GPG signature!" 2>&1; exit 1; fi && \
+  if [[ ${FINGERPRINT} != "${FAIL2BAN_GPG_FINGERPRINT}" ]]; then \
+    echo "ERROR: Wrong GPG fingerprint!" 2>&1; exit 1; fi && \
+  dpkg -i fail2ban_${FAIL2BAN_VERSION}-1.upstream1_all.deb && \
   # cleanup
+  rm -rf fail2ban_${FAIL2BAN_VERSION}-1.upstream1_all.deb && \
```

Comment:
```suggestion
  dpkg -i fail2ban_0.11.2-1.upstream1_all.deb && \
  # cleanup
  rm -rf fail2ban_0.11.2-1.upstream1_all.deb && \
```

Reply:
BTW: I like saving the package just to `fail2ban.deb`:  curl -Lso fail2ban.deb ${FAIL2BAN_URL}`

Then just  `dpkg -i fail2ban.deb` etc. is needed. Looks more clean IMO 😃

Reply:
Added with https://github.com/docker-mailserver/docker-mailserver/commit/3b56e841e64b0081c62c27248a1378ba235b6160